tests/deploy_test.sh

```diff
@@ -697,4 +684,126 @@ function test_cleanup()
   done <<< "$output"
 }
 
+function test_parse_deploy_options()
```

Comment:
Now that you added this test function, I think we can remove all of the tests that rely on `test_mode`, i.e., we can implode `test_kernel_deploy`.